analysis/dataset_definition.py

```diff
@@ -1,14 +1,672 @@
+)
+
+## Chronic obstructive pulmonary disease, on or before baseline
+dataset.cov_bin_copd = (
+    last_matching_event_clinical_snomed_before(copd_snomed_clinical, baseline_date).exists_for_patient() |
+    last_matching_event_apc_before(copd_icd10, baseline_date).exists_for_patient()
+)
+
+## Liver disease, on or before baseline
+dataset.cov_bin_liver_disease = (
+    last_matching_event_clinical_snomed_before(liver_disease_snomed_clinical, baseline_date).exists_for_patient() |
+    last_matching_event_apc_before(liver_disease_icd10, baseline_date).exists_for_patient()
+)
+
+## Chronic kidney disease, on or before baseline
+dataset.cov_bin_chronic_kidney_disease = (
+    last_matching_event_clinical_snomed_before(ckd_snomed_clinical, baseline_date).exists_for_patient() |
+    last_matching_event_apc_before(ckd_icd10, baseline_date).exists_for_patient()
+)
+
+## Gestational diabetes
+dataset.cov_bin_gestationaldm = (
+    last_matching_event_clinical_ctv3_before(diabetes_gestational_ctv3_clinical, baseline_date).exists_for_patient() |
+    last_matching_event_apc_before(gestationaldm_icd10, baseline_date).exists_for_patient()
+)
+
+## PCOS
+dataset.cov_bin_pcos = (
+    last_matching_event_clinical_snomed_before(pcos_snomed_clinical, baseline_date).exists_for_patient() |
+    last_matching_event_apc_before(pcos_icd10, baseline_date).exists_for_patient()
+)
+
+## Type 1 Diabetes
+dataset.cov_bin_t1dm = (
+    last_matching_event_clinical_ctv3_before(diabetes_type1_ctv3_clinical, baseline_date).exists_for_patient() |
+    last_matching_event_apc_before(diabetes_type1_icd10, baseline_date).exists_for_patient()
+)
+
+## Diabetes complications (foot, retino, neuro, nephro)
+dataset.cov_bin_diabetescomp = (
+    last_matching_event_clinical_snomed_before(diabetescomp_snomed_clinical, baseline_date).exists_for_patient() |
+    last_matching_event_apc_before(diabetescomp_icd10, baseline_date).exists_for_patient()
+)
+
+### Any HbA1c measurement
+dataset.cov_bin_hba1c_measurement = last_matching_event_clinical_snomed_before(hba1c_measurement_snomed, baseline_date).exists_for_patient()
+
+### Any OGTT done
+dataset.cov_bin_ogtt_measurement = last_matching_event_clinical_snomed_before(ogtt_measurement_snomed, baseline_date).exists_for_patient()
+
+### Covid-19 vaccination history
+dataset.cov_count_covid_vaccines = (
+  vaccinations
+  .where(vaccinations.target_disease.is_in(["SARS-2 CORONAVIRUS"]))
+  .where(vaccinations.date.is_on_or_before(baseline_date))
+  .count_for_patient()
+)
+dataset.cov_date_recent_covid_vaccines = (
+  vaccinations
+  .where(vaccinations.target_disease.is_in(["SARS-2 CORONAVIRUS"]))
+  .where(vaccinations.date.is_on_or_before(baseline_date))
+  .sort_by(vaccinations.date)
+  .last_for_patient()
+  .date
+)
+
+## BMI, most recent value, within previous 2 years
+bmi_measurement = most_recent_bmi(
+    where=clinical_events.date.is_after(baseline_date - days(2 * 366)),
+    minimum_age_at_measurement=16,
```

Comment:
Do you really need to specify the minimum age of 16? you would exclude the participants aged <18 at baseline anyway in the later pipelines.